src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -29,6 +29,10 @@ private static bool CheckStorageContext(ApplicationEngine engine, StorageContext
                 ContractState contract = engine.Snapshot.Contracts.TryGet(context.ScriptHash);
                 if (contract == null) return false;
                 if (!contract.HasStorage) return false;
+                if (!contract.RedirectionHash.Equals(UInt160.Zero))
```

Comment:
I think it's not suitable to put the change in a check function.

Reply:
Or we can merge this two functions and give a more appropriate name, as both are dealing with `StorageContext`